### PR TITLE
Implement EZP-24804: Ancestor criterion

### PR DESCRIPTION
**Pull request by pspanja**
This PR resolves https://jira.ez.no/browse/EZP-24804
Review together with https://github.com/ezsystems/ezplatform-solr-search-engine/pull/19

This implements `Ancestor` criterion, accepting Location path string(s) as value.

#### TODOs
- [x] implement criterion handlers for Legacy engine
- [x] implement REST input parser

### CODE REVIEW

**Comment by pspanja** [comment]
Ready for review, ping @andrerom @bdunogier @dpobel @glye @lolautruche @yannickroger

**Comment by andrerom** [comment]
No impl for legacy needed?

**Comment by pspanja** [comment]
It is missing :) Todo added.

**Comment by pspanja** [comment]
Implemented handlers for Legacy engine in https://github.com/ezsystems/ezpublish-kernel/commit/5a7f01a4ffa4081566c51388c3e854dafb1ec99b.
Also updated tests to test both EQ and IN operators.

File: eZ/Publish/API/Repository/Tests/SearchServiceTest.php [hunk 1]
```diff
@@ -546,6 +546,16 @@ function ($a, $b) {
                 ),
                 $fixtureDir . 'UserMetadata.php',
             ),
+            array(
+                array(
+                    'filter' => new Criterion\Ancestor(array('/1/5/44/45/')),
```
Comment (andrerom) [inline eZ/Publish/API/Repository/Tests/SearchServiceTest.php]: Maybe change this to test `(array('/1/5/44/', '/1/5/44/45/')` which should produce same result right? _(more logic covered)_

**Comment by andrerom** [comment]
+1 besides possible impr on test.

File: eZ/Publish/API/Repository/Tests/SearchServiceTest.php (hunk 1 above)
Comment (pspanja) [inline eZ/Publish/API/Repository/Tests/SearchServiceTest.php]: True, updated and rebased.

**Comment by lolautruche** [comment]
+1

Don't forget to document it :wink:

**Comment by pspanja** [comment]
Forgot about the REST input parser, todo added.

**Pull request merged**